concepts/infrastructure-reconciliation/README.md

```diff
@@ -1,41 +1,196 @@
+The three new operators would need to be introduced:
+- Gardener Cluster Manager
+- Customer Cluster Manager
+- Subscription Manager
+
+Each of those components has a clear set of responsibilities that together handle the flow that creates a new, complete, Kyma-ready  Kubernetes cluster.
+
+Along the operators, there are four new Custom Resources, that serve as an API for the components:
+- Cluster CR
+- Gardener Cluster CR
+- Customer Cluster CR
+- SubscriptionRequest CR
+
+### Cluster Managers and the Cluster CRs
+
+The Cluster Managers' main responsibility is to manage the respective Cluster CRs, this is tied to both their provisioning and providing an access to the created K8S clusters.
+
+First of all, an overarching Cluster CR will provide a common API for operations connected to support or infrastructure maintenance. It contains reference to the resource corresponding to the sub-type of cluster:
+```
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: Cluster
+metadata:
+	name: managed-cluster
+	namespace: default
+spec:
+	gardenerClusterRef:
+		apiVersion: operator.kyma-project.io/v1alpha1
+		kind: GardenerCluster
+		name: managed-cluster
+		namespace: default
+*** or ***
+    customerClusterRef:
+		apiVersion: operator.kyma-project.io/v1alpha1
+		kind: CustomerCluster
+		name: customer-cluster
+		namespace: default	
+```
+
+Then, there will be two resources, first for the Gardener-based, managed clusters:
+```
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: GardenerCluster
+metadata:
+	name: managed-cluster
+	namespace: default
+spec:
+	core:
+		purpose: "Production"
+		region: "eu-central-1"
+	extensions:
+		shootNetworkingFilterDisabled: true
+	oidcConfig:
+		clientID: "asdf"
+		groupsClaim: "asdf"
+		issuerURL: "asdf"
+		signingAlgs: ["adsf"]
+		usernameClaim: "asdf"
+		usernamePrefix: "asdf"
+	infrastructure:
+		machineType: "m5.2xlarge"
+		autoScalerMin: 5
+		autoScalerMax: 5
+		provider: aws
+		numberOfZones: 3
+	subscriptionSelector: "globalaccount:abcd-0001" || "shared"
+	kubeconfigs:
+	- secretRef: 
+		name: managed-cluster-admin-kubeconfig
+		namespace: default
+	  clusterRole:
+		rules:
+		- apiGroups: ["*"]
+		  resources: ["*"]
+		  verbs: ["*"]
+	administrators: ["user.id@sap.com", "other.id@sap.com"]
+```
+
+This contains a minimal set of values that are required to provision a new Gardener cluster. The resource also contains the `administrators` and `subscriptionSelector` fields that are required for the Binding creation and requesting a hyperscaler subscription respectively.
+
+And the second one for BYOC model:
+```
+customer-cluster
+
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: CustomerCluster
```

Comment:
```suggestion
kind: ExternalCluster
```

Reply:
I think CustomerCluster is not really fitting the role here, but its just a nit, so nothing urgent

Reply:
Yeah, was thinking about the name for so long I've skipped the best (and simple) option. 😄 Will update the CR name to the `ExternalCluster`.